x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -110,30 +118,43 @@ export const createMetricThresholdExecutor = (libs: InfraBackendLibs) =>
       executionId,
     });
 
-    const {
-      alertWithLifecycle,
-      savedObjectsClient,
-      getAlertUuid,
-      getAlertStartedDate,
-      getAlertByAlertUuid,
-    } = services;
+    const { alertsClient, savedObjectsClient } = services;
+    if (!alertsClient) {
```

Comment:
Currently we have the alerts client initialized under undocumented feature flag (which defaults to true), so the alertsClient is typed as possibly undefined. We plan to remove the feature flag so we will be able to remove this check at that point.